romancal/resample/resample.py

```diff
@@ -671,3 +669,85 @@ def drizzle_arrays(
             fillstr=fillval,
         )
+
+
+def l2_into_l3_meta(l3_meta, l2_meta):
+    """Update the level 3 meta with info from the level 2 meta
+
+    Parameters
+    ----------
+    l3_meta : dict
+        The meta to update. This is updated in-place
+
+    l2_meta : stnode
+        The Level 2-like meta to pull from
+
+    Notes
+    -----
+    The list of meta that is pulled from the Level 2 meta into the Level 3 meta is as follows:
+    basic.visit: observation.visit
+    basic.segment: observation.segment
+    basic.pass: observation.pass
+    basic.program: observation.program
+    basic.survey: obervation.survey
+    basic.optical_element: optical_element
+    basic.instrument: instrument.name
+    basic.telescope: telescope
+    program: program
+    """
+    l3_meta.basic.visit = l2_meta.observation.visit
+    l3_meta.basic.segment = l2_meta.observation.segment
+    # l3_meta.basic.pass = l2_meta.observation.pass
+    l3_meta.basic.program = l2_meta.observation.program
+    l3_meta.basic.survey = l2_meta.observation.survey
+    l3_meta.basic.optical_element = l2_meta.instrument.optical_element
+    l3_meta.basic.instrument = l2_meta.instrument.name
+    l3_meta.basic.telescope = l2_meta.telescope
+    l3_meta.program = l2_meta.program
+
+
+def gwcs_into_l3(model, wcsinfo):
+    """Update the Level 3 wcsinfo block from a GWCS object
+
+    Parameters
+    ----------
+    model : `DataModel`
+        The model whose meta is to be updated.
+
+    wcsinfo : `GWCS`
+        GWCS info to transfer into the `meta.wcsinfo` block
+
+    Notes
+    -----
+    Some models/parameters in the GWCS object have explicit names, such as
+    'crpix1'. However, some do not and hence have to be accessed explicitly
+    by indexing. This is fragile and will be a source of issues.
+    """
+    l3_wcsinfo = model.meta.wcsinfo
+    transform = wcsinfo.forward_transform
+
+    # Basic WCS info
+    l3_wcsinfo.projection = "TAN"
+    l3_wcsinfo.rotation_matrix = transform["pc_rotation_matrix"].matrix.value.tolist()
+    l3_wcsinfo.dec_ref = transform.lat_6.value
+    l3_wcsinfo.ra_ref = transform.lon_6.value
+    # l3_wcsinfo.x_ref = center of mosaic?
+    # l3_wcsinfo.y_ref = center of mosaic?
+    # l3_wcsinfo.s_region = from gwcs.bounding_box
+
+    # Mosaic info
+    # l3_wcsinfo.dec_center = from center of mosaic
+    # l3_wcsinfo.pixel_scale = ???
+    # l3_wcsinfo.pixel_scale_local = ???
+    # l3_wcsinfo.pixel_shape = image shape
+    # l3_wcsinfo.ra_center = from center of mosaic
+    # l3_wcsinfo.ra_corn1 = from bounding box
+    # l3_wcsinfo.ra_corn2 = from bounding box
+    # l3_wcsinfo.ra_corn3 = from bounding box
+    # l3_wcsinfo.ra_corn4 = from bounding box
+    # l3_wcsinfo.dec_corn1 = from bounding box
+    # l3_wcsinfo.dec_corn2 = from bounding box
+    # l3_wcsinfo.dec_corn3 = from bounding box
+    # l3_wcsinfo.dec_corn4 = from bounding box
+    # l3_wcsinfo.orientat = ???
+    # l3_wcsinfo.orientat_local = ???
```

Comment:
I guess we can fill these out in a later PR.  You might see if you can fill out some of them using the information in this file on central store, roman/eschlafly/tessellation/tessellation.py, though I am coming a bit from the opposite direction (trying to make a gwcs object) to some extent.